cmd/bnsd/client/client.go

```diff
@@ -392,6 +394,33 @@ func walletKeyToAddr(key []byte) weave.Address {
 	return key[5:]
 }
 
+type CurrenciesResponse struct {
+	Height     int64
+	Currencies map[string]currency.TokenInfo
+}
+
+func (b *BnsClient) ListCurrencies() (*CurrenciesResponse, error) {
+	resp, err := b.AbciQuery("/tokens?prefix", nil)
+	if err != nil {
+		return nil, err
+	}
+	if len(resp.Models) == 0 { // empty list or nil
+		return nil, nil
+	}
+	out := &CurrenciesResponse{
+		Height:     resp.Height,
+		Currencies: make(map[string]currency.TokenInfo),
+	}
+	for _, v := range resp.Models {
+		var ti currency.TokenInfo
+		if err := ti.Unmarshal(v.Value); err != nil {
+			return nil, err
```

Comment:
I think it's better to wrap the error, so that it's easier to track where it comes from.